src/group/mls_group/api.rs

```diff
@@ -17,20 +17,22 @@
 use crate::framing::*;
 use crate::group::*;
 use crate::key_packages::*;
-use crate::messages::*;
-pub trait Api {
+use crate::messages::{proposals::*, *};
+use crate::tree::{index::LeafIndex, node::*};
+
+pub trait Api: Sized {
     /// Create a new group.
     fn new(
         group_id: &[u8],
         ciphersuite: Ciphersuite,
-        key_package_bundle: (HPKEPrivateKey, KeyPackage),
+        key_package_bundle: KeyPackageBundle,
```

Comment:
So far I avoided `KeyPackageBundle` in the API, because it introduces a new type that doesn't exist in the spec. But I agree that it looks better than a tuple.

Reply:
Yeah I was wondering about that, what's the best way is to handle this.
For this generic API having two separate arguments for key and key package might be the best.